system/scripts/zone/core/calc_companion.cs

```diff
@@ -0,0 +1,394 @@
+﻿//--- Melia Script ----------------------------------------------------------
+// Companion Calculation Script
+//--- Description -----------------------------------------------------------
+// Functions that calculate companion-related values, such as properties.
+//---------------------------------------------------------------------------
+
+using System;
+using Melia.Shared.Game.Const;
+using Melia.Zone.Scripting;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Monsters;
+using Melia.Zone.World.Maps;
+
+public class CompanionCalculationsScript : GeneralScript
+{
+	private const float PET_STAT_BY_OWNER_RATE = 0.5f;
+
+	public float PET_STAT_BY_OWNER(Companion companion, string statName)
```

Comment:
I assume that you're imitating official names with the constant and this function, but generally speaking, using upper case here goes entirely against our normal naming conventions, and we presumably also won't need these for reference. As such, the normal PascalCase would be more appropriate.